lib/elixir/lib/list.ex

```diff
@@ -591,15 +591,21 @@ defmodule List do
   """
   @spec to_string(:unicode.charlist) :: String.t
   def to_string(list) when is_list(list) do
-    case :unicode.characters_to_binary(list) do
-      result when is_binary(result) ->
-        result
-
-      {:error, encoded, rest} ->
-        raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :invalid
-
-      {:incomplete, encoded, rest} ->
-        raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :incomplete
+    try do
+      case :unicode.characters_to_binary(list) do
+        result when is_binary(result) ->
+          result
+
+        {:error, encoded, rest} ->
+          raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :invalid
+
+        {:incomplete, encoded, rest} ->
+          raise UnicodeConversionError, encoded: encoded, rest: rest, kind: :incomplete
+      end
+    rescue
+      ArgumentError ->
+        # Instead of a more cryptic ":unicode.characters_to_binary" error.
```

Comment:
It does mean we'll do the characters_to_binary work twice in happy cases – is that acceptable?